integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/backend.py

```diff
@@ -1,6 +1,24 @@
 from abc import ABC, abstractmethod
 from typing import Any, Dict, List, Optional, Tuple
 
+from pydantic import BaseModel, ConfigDict
+
+
+class Model(BaseModel):
```

Comment:
This introduces a dependency on `pydantic`, but it looks like we can implement this with a regular `dataclass`?